pinakes/main/approval/views.py

```diff
@@ -313,22 +313,38 @@ def unlink(self, request, pk):
         )
         return Response(status=status.HTTP_204_NO_CONTENT)
 
-    def perform_create(self, serializer):
-        max_obj = (
-            Workflow.objects.filter(tenant=Tenant.current())
-            .order_by("-internal_sequence")
-            .first()
-        )
-        if max_obj is None:
-            next_seq = Decimal(1)
-        else:
-            next_seq = Decimal(
-                max_obj.internal_sequence.to_integral_value() + 1
+    @extend_schema(
+        request=RepositionSerializer,
+        responses={204: None},
+    )
+    @action(methods=["post"], detail=True)
+    def reposition(self, request, pk):
+        """
+        Adjust the position of a workflow related to others by an offset number
+        """
+        serializer = RepositionSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(
+                {"errors": serializer.errors},
+                status=status.HTTP_400_BAD_REQUEST,
             )
-        serializer.save(
-            internal_sequence=next_seq,
-            tenant=Tenant.current(),
-        )
+
+        if "placement" in serializer.validated_data:
+            if serializer.validated_data["placement"] == "top":
+                offset = -math.inf
```

Comment:
Constants `math.inf` are still used to determine first \ last position here.

Reply:
See my comments above. We don't need a constant, just need some big number.